mockgcp/mockcontainer/cluster.go

```diff
@@ -511,6 +525,10 @@ func (s *ClusterManagerV1) populateClusterDefaults(obj *pb.Cluster) error {
 		obj.MonitoringService = "monitoring.googleapis.com/kubernetes"
 	}
 
+	if obj.NetworkPolicy != nil && obj.NetworkPolicy.String() == "" {
+		obj.NetworkPolicy = nil
+	}
+
```

Comment:
is this closer to how the real API behaves?

Reply:
Yes. When passing [`networkPolicy: {}` in POST](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/a30525b20ea4741bd9cc3918df785410e01c59f0/pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling/_http.log#L87), [the retrieved cluster doesn't contain `networkPolicy` field in the response](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/a30525b20ea4741bd9cc3918df785410e01c59f0/pkg/test/resourcefixture/testdata/basic/container/v1beta1/containercluster/containercluster-autoscaling/_http.log#L194).